event-asynchronous/src/test/java/com/iluwatar/event/asynchronous/EventAsynchronousTest.java

```diff
@@ -0,0 +1,82 @@
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.iluwatar.event.asynchronous;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ *
+ * Application test
+ *
+ */
+public class EventAsynchronousTest {
+  App app;
+
+  @Before
+  public void setUp() {
+    app = new App();
+  }
+
+  @Test
+  public void testAsynchronousEvent() {
+    EventManager eventManager = new EventManager();
+    try {
+      int aEventId = eventManager.createAsync(60);
+      eventManager.start(aEventId);
+      assertTrue(eventManager.getEventPool().size() == 1);
+      assertTrue(eventManager.getEventPool().size() < EventManager.MAX_RUNNING_EVENTS);
+      assertTrue(eventManager.numOfCurrentlyRunningSyncEvent() == -1);
+      eventManager.cancel(aEventId);
+      assertTrue(eventManager.getEventPool().size() == 0);
+    } catch (MaxNumOfEventsAllowedException | LongRunningEventException | EventDoesNotExistException e) {
+      System.out.println(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testSynchronousEvent() {
+    EventManager eventManager = new EventManager();
+    try {
+      int sEventId = eventManager.create(60);
+      eventManager.start(sEventId);
+      assertTrue(eventManager.getEventPool().size() == 1);
+      assertTrue(eventManager.getEventPool().size() < EventManager.MAX_RUNNING_EVENTS);
+      assertTrue(eventManager.numOfCurrentlyRunningSyncEvent() != -1);
+      eventManager.cancel(sEventId);
+      assertTrue(eventManager.getEventPool().size() == 0);
+    } catch (MaxNumOfEventsAllowedException | LongRunningEventException | EventDoesNotExistException
+        | InvalidOperationException e) {
+      System.out.println(e.getMessage());
+    }
+  }
+
+  @Test(expected = InvalidOperationException.class)
+  public void testUnsuccessfulSynchronousEvent() throws InvalidOperationException {
+    EventManager eventManager = new EventManager();
+    try {
+      int sEventId = eventManager.create(60);
+      eventManager.start(sEventId);
+      sEventId = eventManager.create(60);
+      eventManager.start(sEventId);
+    } catch (MaxNumOfEventsAllowedException | LongRunningEventException | EventDoesNotExistException e) {
+      System.out.println(e.getMessage());
+    }
+  }
```

Comment:
We should add tests to run operations from start to finish also. Create and start event and wait for it to finish. For async operations we could create multiple events and wait for all of them to finish.